### PR TITLE
NEW: Add tests for default UI input actions (ISX-1915).

### PR DESCRIPTION
**Pull request by graham-huws**
### Description

Extend tests to check all default UI actions work with and without a set project-wide asset - [ISX-1915](https://jira.unity3d.com/browse/ISX-1915).

### Changes made

As above

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs [hunk 1]
```diff
@@ -118,7 +101,7 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
         Move(mouse.position, new Vector2(100f, 0.5f));
         Update();
 
-        Assert.IsTrue(m_InputForUIEvents.Count == 1);
+        Assert.AreEqual(1, m_InputForUIEvents.Count);
```
Comment (ekcoh) [inline Assets/Tests/InputSystem/Plugins/InputForUITests.cs:112]: I was not part of establishing the convention so do not now why, but Assert.That(m_InputForUIEvents.Count, Is.EqualTo(1)); seems preferred instead. Likely due to not mixing up expected and actual argument order...

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs [hunk 2]
```diff
@@ -127,78 +110,389 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
...
+    // When there are no project-wide actions asset, the InputSystemProvider should still work as it currently gets
+    // the actions from DefaultActionsAsset().asset.
+
     [Test]
     [Category(kTestCategory)]
-    // Presses a gamepad left stick left and verifies that a navigation move event is dispatched
-    public void NavigationMoveWorks()
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_NavigationActionsWork(bool useProjectWideActionsAsset)
```
Comment (ekcoh) [inline Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: What does "work" mean? I would suggest renaming this test to provide more clarity, e.g. "NavigationActionFiresUINavigationEvent_RegardlessIfInputIsGamepadJoystickOrKeyboard"

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs [hunk 3]
```diff
@@ -127,78 +110,389 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
...
+        }
+
+
+        var joystick = InputSystem.AddDevice<Joystick>();
+        PressUpdateReleaseUpdate(joystick.stick.left);
+        PressUpdateReleaseUpdate(joystick.stick.up);
+        PressUpdateReleaseUpdate(joystick.stick.right);
+        PressUpdateReleaseUpdate(joystick.stick.down);
+
+        Assert.That(m_InputForUIEvents[12] is
```
Comment (ekcoh) [inline Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: All of these specific offsets into event buffer, how likely do you think they are to change and break this test? I notice we do not check in-between events, would it make more sense to assert in-between each actuation above?

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs [hunk 4]
```diff
@@ -127,78 +110,389 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
...
+            asPointerEvent: { type: PointerEvent.Type.PointerMoved, eventSource: EventSource.Touch }
+        });
     }
 
     [Test]
     [Category(kTestCategory)]
-    public void SendWheelEvent()
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_ClickActionWorks(bool useProjectWideActionsAsset)
```
Comment (ekcoh) [inline Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: @graham-huws did you notice any difference in behavior on these clicks? I see left/middle/right is configured differently in the PWA asset

**Review by ekcoh — APPROVED** [review]
In general looks good to me, some minor comments. Thanks for improving the test coverage on the UI integration.

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs (hunk 1 above)
Comment (graham-huws-unity) [reply Assets/Tests/InputSystem/Plugins/InputForUITests.cs:112]: Oh, interesting - I assumed everyone was using Rider which shows the arguments:
image
Though I guess github lacks that 😄

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs (hunk 3 above)
Comment (graham-huws-unity) [reply Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: Good points, I'll look into this

File: Assets/Tests/InputSystem/Plugins/InputForUITests.cs (hunk 4 above)
Comment (graham-huws-unity) [reply Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: Not sure what you mean here sorry - They produce the different PointerEvent.Button (left/middle/right), is there something else I should be checking for?

Comment (ekcoh) [reply Assets/Tests/InputSystem/Plugins/InputForUITests.cs]: Sorry for vagueness, I was basically referring to the fact that they have different configuration...

Click has initialStateCheck = true while Middle and Right Clicks have initialStateCheck = false, doesn't make sense to me....

**Pull request merged**